snforge_std/src/cheatcodes.cairo

```diff
@@ -179,3 +185,39 @@ fn get_class_hash(contract_address: ContractAddress) -> ClassHash {
     let buf = cheatcode::<'get_class_hash'>(array![contract_address_felt].span());
     (*buf[0]).try_into().expect('Invalid class hash value')
 }
+
+trait L1Handler {
+    fn build(contract_address: ContractAddress, selector_name: felt252) -> L1HandlerFn;
+    fn invoke(self: L1HandlerFn, from_address: felt252, fee: u128, payload: Span::<felt252>);
+}
+
+impl L1HandlerImpl of L1Handler {
+    fn build(contract_address: ContractAddress, selector_name: felt252) -> L1HandlerFn {
```

Comment:
At the first time using this I was like "why using build function instead of init a struct", but @MaksymilianDemitraszek you were already thinking about "what if we want to do some processing during the build", right? Which.. well, sounds pretty nice and the interface is still elegant IMO.

Reply:
@MaksymilianDemitraszek I removed the `build` from the trait as it forced someone to directly init from the trait `L1HandlerTrait::build(..)` which look like less natural. For instance the dispatchers for `MockCall` are not built with a `build` function but directly instantiating the struct.

Is this interface ok for you then?

```rust
let l1_handler = L1Handler {
    contract_address,
    selector_name: 'process_l1_message'
};

l1_handler.invoke(0x123, 1, payload.span());
```

Reply:
Why build forces you to do that? Imo you can provide `default` as a simplified interface and if somebody needs more customization then they can create the structure by hand.

Reply:
Answered in other related comment. :+1: